spacewalk/setup/lib/Spacewalk/Setup.pm

```diff
@@ -880,11 +880,41 @@ sub postgresql_reportdb_setup {
 	postgresql_drop_reportdb($answers);
     }
 
-    $ENV{PGSSLROOTCERT} = $answers->{'report-db-ca-cert'};
+    if ($answers->{'external-db'}) {
+        $ENV{PGSSLROOTCERT} = $answers->{'external-db-ca-cert'};
+    }
+    else {
+        $ENV{PGSSLROOTCERT} = $answers->{'report-db-ca-cert'};
+    } 
     $ENV{PGSSLMODE} = "verify-full";
 
     if ($answers->{'report-db-host'} ne '') {
             write_rhn_conf($answers, 'report-db-backend', 'report-db-host', 'report-db-port', 'report-db-name', 'report-db-user', 'report-db-password', 'report-db-ssl-enabled');
+            if ($answers->{'external-db'}) {
+              my @cmd = ("/usr/bin/uyuni-setup-reportdb",
+                    "create_remote",
+                    "--db", $answers->{'report-db-name'},
+                    "--external-host", $answers->{'report-db-host'},
+                    "--external-user", $answers->{'external-db-user'},
+                    "--external-password", $answers->{'external-db-password'},
+                    "--user", $answers->{'report-db-user'});
+              if ($answers->{'report-db-password'} ne '') {
+                      push @cmd, "--password", $answers->{'report-db-password'};
+              }
+              else {
+                      push @cmd, "--autogenpw";
+              }
+              if ($answers->{'external-db-provider'} ne '') {
+                      push @cmd, "--external-provider", $answers->{'external-db-provider'};
+              }
+
+              print_progress(-init_message => "*** Progress: #",
+                      -log_file_name => DB_INSTALL_LOG_FILE,
+                      -log_file_size => DB_INSTALL_LOG_SIZE,
+                      -err_message => "Could not install report database.\n",
+                      -err_code => 15,
+                      -system_opts => \@cmd);
+            }          
```

Comment:
All this duplicate a lot of code. Here in the setup and also in the tool itself.
Can't this be joined? 
- keep just `create` and `remove` command.
- if you mean external DB, provide the extra options. 
- In the tool provide for every command the local and the external variant and select it depending on what options were provided.

Reply:
it is a lot of code. The other option is add a huge if block to the create in uyuni-setuo-reportdb.  You can see here https://github.com/uyuni-project/uyuni/blob/master/spacewalk/uyuni-setup-reportdb/bin/uyuni-setup-reportdb#L119  this function is 99% seting up the local postgres. I am fine either way

Reply:
I cleaned this up in Setup.pm as well now.